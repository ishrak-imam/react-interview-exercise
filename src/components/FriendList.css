:local(.friendList) {
  padding-left: 0;
  margin-bottom: 0;
  margin-top: 10px
}
