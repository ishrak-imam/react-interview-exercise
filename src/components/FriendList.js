import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './FriendList.css'
import FriendListItem from './FriendListItem'

class FriendList extends Component {
  render () {
    const { friendIds } = this.props
    return (
      <ul className={styles.friendList}>
        {friendIds.map(id => {
          return (
            <FriendListItem key={id} friendId={id} />
          )
        })}
      </ul>
    )
  }
}

FriendList.propTypes = {
  friendIds: PropTypes.array.isRequired
}

export default FriendList
