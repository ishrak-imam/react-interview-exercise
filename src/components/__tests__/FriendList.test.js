import React from 'react'
import { create } from 'react-test-renderer'
import FriendList from '../FriendList'

test('FriendList component', () => {
  const component = create(
    <FriendList friendIds={[]} deleteFriend={() => {}} starFriend={() => {}} />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
