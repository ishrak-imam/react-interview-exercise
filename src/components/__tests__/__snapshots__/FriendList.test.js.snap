exports[`test FriendList component 1`] = `
<ul
  className={undefined} />
`;
